app/src/main/java/sdp/moneyrun/menu/JoinGameImplementation.java

```diff
@@ -89,32 +96,90 @@ private void onJoinGamePopupWindowLoadGameList(@NonNull PopupWindow popupWindow)
             EditText filterEditText = popupWindow.getContentView().findViewById(R.id.join_game_text_filter);
             openGamesLayout.removeAllViews();
 
-            String filterText = filterEditText.getText().toString().trim().toLowerCase(Locale.getDefault());
-            loadGameListGivenFilter(popupWindow, openGamesLayout, filterText);
+            filterText = filterEditText.getText().toString().trim().toLowerCase(Locale.getDefault());
+            loadGameListGivenFilter(popupWindow, openGamesLayout);
         });
 
         // First load the game list without any filter.
-        loadGameListGivenFilter(popupWindow, openGamesLayout, null);
+        loadGameListGivenFilter(popupWindow, openGamesLayout);
     }
 
-    private void loadGameListGivenFilter(@NonNull PopupWindow popupWindow, @NonNull LinearLayout openGamesLayout, @Nullable String filterText) {
+    @SuppressLint("MissingPermission")
+    private void loadGameListGivenFilter(@NonNull PopupWindow popupWindow,
+                                         @NonNull LinearLayout openGamesLayout) {
         List<GameRepresentation> gameRepresentations = new ArrayList<>();
         Task<DataSnapshot> taskDataSnapshot = getTaskGameRepresentations(gameRepresentations);
         taskDataSnapshot.addOnSuccessListener(dataSnapshot -> {
             TableLayout gameLayout = new TableLayout(activity);
 
-            buttonId = 0;
-            for (GameRepresentation gameRepresentation : gameRepresentations) {
-                String lowerName = gameRepresentation.getName().toLowerCase(Locale.getDefault());
-                if (filterText == null || lowerName.contains(filterText)) {
-                    displayGameInterface(popupWindow, gameLayout, buttonId, gameRepresentation);
-                    buttonId++;
-                }
-            }
+            // Grant permissions if necessary
+            requestLocationPermissions(requestPermissionsLauncher);
+
+            loadGameListFromLocation(gameRepresentations, popupWindow, gameLayout, false);
             openGamesLayout.addView(gameLayout);
+
+            fusedLocationClient.getLastLocation()
+                    .addOnSuccessListener(activity, location -> {
+                        // Got last known location. In some rare situations this can be null
+                        // In this case, the game cannot be instantiated
```

Comment:
Could you put this in a function and test it ?

Reply:
I already tried putting this in a function, but there is too many dependencies on other methods. I think the best idea would be to get an abstract location service to be able to mock the location, not wrapping half of the activity in a method and then test it.

Reply:
Ok I'll let you do that then

Reply:
This is not the purpose of this branch, I think it would be better to define a new task realted to that.